tests/strategies/test_bagua_strategy.py

```diff
@@ -0,0 +1,51 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import pytest
+import torch
+
+from pytorch_lightning import Trainer
+from pytorch_lightning.strategies import BaguaStrategy
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from tests.helpers.boring_model import BoringModel
+from tests.helpers.runif import RunIf
+
+
+class TestModel(BoringModel):
+    def __init__(self):
+        super().__init__()
+        self.layer = torch.nn.Linear(32, 2)
+
+
+@RunIf(skip_windows=True, bagua=True, min_gpus=2, standalone=True)
+def test_bagua_algorithm(tmpdir):
+    model = TestModel()
+    bagua_strategy = BaguaStrategy(algorithm="gradient_allreduce")
```

Comment:
Can we add a test parametrizing all algorithms and testing that bagua is instantiated as expected? It does not need to `fit`. You could mock `BaguaDistributedDataParallel`

Reply:
Yes, We tested for all algorithms previously, but it turns out  it is quite time consuming. After discussed with @tchaton , we decided to move other tests to Bagua side and only keep a simple gut check in lightning.

Reply:
Yes. Bagua has this kind of validation, it will throw an exception on invalid inputs.

Reply:
I am referring to tests that check that we pass the expected information to Bagua.

Reply:
I added a few tests to check this.